subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -109,124 +109,136 @@ static int update_job_execution(struct mqtt_client *const client,
-						      update_topic);
+	} else if (execution_state == AWS_JOBS_IN_PROGRESS &&
+			fota_state == APPLY_UPDATE) {
+		LOG_INF("Firmware download completed");
+		execution_state = AWS_JOBS_SUCCEEDED;
+		err = update_job_execution(client, job_id,
+				execution_state, fota_state,
+				doc_version_number, "");
 		if (err) {
-			LOG_ERR("Error when subscribing job_id_update: "
-				"%d", err);
 			return err;
 		}
+	} else if (execution_state == AWS_JOBS_SUCCEEDED &&
+			fota_state == APPLY_UPDATE) {
+		LOG_INF("Job document updated with SUCCEDED");
+		LOG_INF("Ready to reboot");
+		callback(AWS_FOTA_EVT_DONE);
+	}
+	return 1;
+}
 
-		/* Set fota_state to DOWNLOAD_FIRMWARE, when we are subscribed
-		 * to job_id topics we will try to publish and if accepted we
-		 * can start the download
-		 */
-		fota_state = DOWNLOAD_FIRMWARE;
+static int aws_fota_on_publish_evt(struct mqtt_client *const client,
+				   const u8_t *topic,
+				   u32_t topic_len,
+				   u32_t payload_len)
+{
+	int err;
 
-		/* Handled by the library */
-		return 1;
+	LOG_INF("Received topic: %s", log_strdup(topic));
 
-	} else if (aws_jobs_cmp(update_topic, topic, topic_len, "accepted")) {
-		LOG_DBG("Job document update was accepted");
-		err = get_published_payload(client, payload_buf, payload_len);
-		if (err) {
-			return err;
-		}
-		/* Update accepted, increment document version counter. */
-		doc_version_number++;
-
-		if (fota_state == DOWNLOAD_FIRMWARE) {
-			/*Job document is updated and we are ready to download
-			 * the firmware.
-			 */
-			execution_state = AWS_JOBS_IN_PROGRESS;
-			LOG_INF("Start downloading firmware from %s%s",
-				log_strdup(hostname), log_strdup(file_path));
-			err = fota_download_start(hostname, file_path);
-			if (err) {
-				LOG_ERR("Error when trying to start firmware"
-				       "download: %d", err);
-				return err;
-			}
-		} else if (execution_state == AWS_JOBS_IN_PROGRESS &&
-		    fota_state == APPLY_UPDATE) {
-			LOG_INF("Firmware download completed");
-			execution_state = AWS_JOBS_SUCCEEDED;
-			err = update_job_execution(client, job_id,
-						   execution_state, fota_state,
-						   doc_version_number, "");
-			if (err) {
-				return err;
-			}
-		} else if (execution_state == AWS_JOBS_SUCCEEDED &&
-			   fota_state == APPLY_UPDATE) {
-			LOG_INF("Job document updated with SUCCEDED");
-			LOG_INF("Ready to reboot");
-			callback(AWS_FOTA_EVT_DONE);
-		}
-		return 1;
-	} else if (aws_jobs_cmp(update_topic, topic, topic_len, "rejected")) {
+	bool is_get_next_topic = aws_jobs_cmp(get_topic, topic, topic_len, "");
+	bool is_get_next_accepted = aws_jobs_cmp(get_topic, topic, topic_len,
+						 "accepted");
+	bool is_notify_next_topic = aws_jobs_cmp(notify_next_topic, topic,
+						 topic_len, "");
+	bool doc_update_accepted = aws_jobs_cmp(update_topic, topic, topic_len,
+					       "accepted");
+	bool doc_update_rejected = aws_jobs_cmp(update_topic, topic, topic_len,
+						"rejected");
```

Comment:
```suggestion
						
	bool is_get_next_topic =
		aws_jobs_cmp(get_topic, topic, topic_len, "");

	bool is_get_next_accepted =
		aws_jobs_cmp(get_topic, topic, topic_len, "accepted");

	bool is_notify_next_topic =
		aws_jobs_cmp(notify_next_topic, topic, topic_len, "");

	bool doc_update_accepted =
		aws_jobs_cmp(update_topic, topic, topic_len, "accepted");

	bool doc_update_rejected =
		aws_jobs_cmp(update_topic, topic, topic_len, "rejected");


```

Reply:
Just a proposal. For me this style makes it easier to read :)

Reply:
Did this, I don't like how it looks but I'm not against it :)